Issues placing orders on singular samples of trio cases

## Description

A customer attempted to place a re-run order on a single sample out of a trio. This created a ticket however the case and order was not properly placed. When investigating the logs I found that:

```shell
2021-03-10 10:17:32 clinical-db.scilifelab.se cg.server.api[44910] INFO processing 'mip-dna' order: name='order_name' comment=None customer='cust002' samples=[{'internal_id': 'Internal_sample_ID', 'name': 'customer_sample_ID', 'data_
analysis': 'mip-dna', 'data_delivery': 'scout', 'application': 'WGTPCFC030', 'family_name': 'customer_case_ID', 'sex': 'male', 'source': 'other', 'priority': 'standard', 'formalin_fixation_time': None, 'post_formalin_fixation_time': Non
e, 'tissue_block_size': None, 'age_at_sampling': 'age_float', 'cohorts': [''], 'phenotype_terms': [''], 'synopsis': [''], 'comment': 'customer_case_ID Data finns customer_sample_ID', 'tumour': False, 'tumour_purity': None, 'capture_kit': None, 'c
ontainer': 'Tube', 'container_name': 'container_ID', 'well_position': None, 'volume': None, 'quantity': '', 'require_qcok': False, 'elution_buffer': None, 'panels': ['SKD'], 'status': 'affected', 'mother': 'mother_cust_ID', 'father':
 'father_cust_OD'}]
2021-03-10 10:17:33 clinical-db.scilifelab.se cg.meta.orders.ticket_handler[44910] INFO Could not detected ticket number in name order_name
2021-03-10 10:17:33 clinical-db.scilifelab.se urllib3.connectionpool[44910] DEBUG Starting new HTTPS connection (1): clinical-scilifelab.supportsystem.com:443
2021-03-10 10:18:01 clinical-db.scilifelab.se urllib3.connectionpool[44910] DEBUG https://clinical-scilifelab.supportsystem.com:443 "POST /api/tickets.json HTTP/1.1" 201 6
2021-03-10 10:18:01 clinical-db.scilifelab.se cg.meta.orders.ticket_handler[44910] INFO 689193: opened new ticket
2021-03-10 10:18:01 [44910] [ERROR] Error handling request
Traceback (most recent call last):
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/gunicorn-19.1.0-py3.6.egg/gunicorn/workers/sync.py", line 93, in handle
    self.handle_request(listener, req, client, addr)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/gunicorn-19.1.0-py3.6.egg/gunicorn/workers/sync.py", line 134, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/cg/server/api.py", line 72, in submit_order
    user_mail=g.current_user.email,
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/cg/meta/orders/api.py", line 60, in submit
    return order_func(order)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/cg/meta/orders/api.py", line 134, in _submit_mip_dna
    return self._submit_case_samples(order)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/cg/meta/orders/api.py", line 120, in _submit_case_samples
    result = self._process_case_samples(order)
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/cg/meta/orders/api.py", line 183, in _process_case_samples
    cases=status_data["families"],
  File "/home/hiseq.clinical/miniconda/envs/prod170926/lib/python3.6/site-packages/cg/meta/orders/status.py", line 257, in store_cases
    mother_obj = family_samples[sample["mother"]] if sample.get("mother") else None
KeyError: 'mother_cust_ID'
```

## Suggested solution

It seems like the order is trying to fetch the mother from the new case, however the sample was ordered as a singlet thus no mother is connected to it.

## This can be closed when

Customers can order a re-run of a single sample from a trio case.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,10 @@ Try to use the following format:
 ### Changed
 ### Fixed
 
+## [NG.NG.NG]
+### Fixed
+- Bugfix allowing orders of single samples from existing families
+
 ## [20.17.4]
 ### Added
 - Added new covid prep method to lims constants

--- a/cg/meta/orders/status.py
+++ b/cg/meta/orders/status.py
@@ -274,8 +274,8 @@ class StatusHandler:
                     self.status.add(new_delivery)
 
             for sample in case["samples"]:
-                mother_obj = family_samples[sample["mother"]] if sample.get("mother") else None
-                father_obj = family_samples[sample["father"]] if sample.get("father") else None
+                mother_obj = family_samples.get(sample["mother"]) if sample.get("mother") else None
+                father_obj = family_samples.get(sample["father"]) if sample.get("father") else None
                 with self.status.session.no_autoflush:
                     link_obj = self.status.link(case_obj.internal_id, sample["internal_id"])
                 if link_obj:

--- a/tests/meta/orders/test_meta_orders_status.py
+++ b/tests/meta/orders/test_meta_orders_status.py
@@ -662,3 +662,55 @@ def test_store_cancer_samples(orders_api, base_store, balsamic_status_data):
     assert base_store.deliveries().count() == base_store.samples().count()
     for link in new_case.links:
         assert len(link.sample.deliveries) == 1
+
+
+def test_store_existing_single_sample_from_trio(orders_api, base_store, mip_status_data):
+
+    # GIVEN a stored trio case
+    new_families = orders_api.store_cases(
+        customer=mip_status_data["customer"],
+        order=mip_status_data["order"],
+        ordered=dt.datetime.now(),
+        ticket=123456,
+        cases=mip_status_data["families"],
+    )
+
+    new_case = new_families[0]
+    assert new_case.name == "family1"
+    assert set(new_case.panels) == {"IEM"}
+    assert new_case.priority_human == "standard"
+
+    assert len(new_case.links) == 3
+    new_link = new_case.links[0]
+    assert new_link.mother
+    assert new_link.father
+    name = new_link.sample.name
+    internal_id = new_link.sample.internal_id
+    assert base_store.sample(internal_id)
+
+    # WHEN storing a new case with one sample from the trio
+    for family in mip_status_data["families"]:
+        for sample_idx, sample in enumerate(family["samples"]):
+            if sample["name"] == name:
+                sample["internal_id"] = internal_id
+                family["name"] = "single-from-trio"
+            else:
+                family["samples"][sample_idx] = {}
+
+        family["samples"] = list(filter(None, family["samples"]))
+
+    print(mip_status_data)
+
+    new_families = orders_api.store_cases(
+        customer=mip_status_data["customer"],
+        order=mip_status_data["order"],
+        ordered=dt.datetime.now(),
+        ticket=123456,
+        cases=mip_status_data["families"],
+    )
+
+    # THEN there should be no complaints about missing parents
+    assert len(new_families) == 1
+    assert len(new_families[0].links) == 1
+    assert not new_families[0].links[0].mother
+    assert not new_families[0].links[0].father

